pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -341,6 +344,46 @@ public SuccessResponse updateTableConfig(
 
     return new SuccessResponse("Table config updated for " + tableName);
   }
+  
+  @POST
```

Comment:
Can we please add documentation highlighting the differences between this and table drop?

Generally, the standard implemented by other databases has the following distinction:

DROP will delete table data, indexes, metadata etc. So any statement (to fetch data or query metadata) will fail stating "table does not exist"

TRUNCATE will delete table data, indexes, but the metadata will be kept around. So you can still see table's schema, table config etc and the queries will return empty results AFAIK.

Are we following the same distinction? If so, we should update the docs to state when should the user choose DROP vs TRUNCATE.

Also, we should add tests to ensure that both behave expectedly. If TRUNCATE is not expected to delete config and schema, then we should add a test for that.

Reply:
+1. drop and truncate behavior should follow the standards.